torch/onnx/_internal/fx/export.py

```diff
@@ -500,6 +559,184 @@ def _validate_op_between_ort_torch(
         warnings.warn(f"ORT fails to run with error: {e}.")
 
 
+def _location_from_fx_stack_trace(
+    node_stack_trace: str,
+) -> Optional[diagnostics.infra.Location]:
+    """Extract location from FX node stack trace.
+
+    Args:
+        node_stack_trace: The stack trace of the FX node. Example:
+
+            File "path/file.py", line 311, in <function>
+                <code>
+            |   File "path/file2.py", line 389, in <function>
+                <code>
+
+    Returns:
+        location: The location of the FX node.
+    """
+    lines = node_stack_trace.strip().split("\n")
+    idx = 0
+    while idx < len(lines) and "File" not in lines[idx]:
```

Comment:
Can we check "File" existence before split it with "\n" to avoid this while loop?

Reply:
Hmm ok to do some sanity checks.